docs/src/pages/guides/routing/LinkRouterWithTheme.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { MemoryRouter as Router } from 'react-router';
+import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
+import { Theme, ThemeProvider, createMuiTheme } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import Link from '@material-ui/core/Link';
+
+const LinkBehavior = React.forwardRef<
+  any,
+  Omit<RouterLinkProps, 'to'> & { href: RouterLinkProps['to'] }
+>((props, ref) => {
+  const { href, ...other } = props;
+  return <RouterLink data-testid="custom-link" ref={ref} to={href} {...other} />;
+});
+
+const themeSetter = (outerTheme: Theme) =>
+  createMuiTheme(outerTheme, {
+    components: {
+      MuiLink: {
+        defaultProps: {
+          // @ts-ignore
+          component: LinkBehavior,
+        },
+      },
+      MuiButtonBase: {
+        defaultProps: {
+          LinkComponent: LinkBehavior,
+        },
+      },
+    },
+  });
+
+export default function LinkRouterWithTheme() {
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        typography: 'body1',
+        flexDirection: 'column',
+        alignItems: 'center', // TODO Replace with Stack
```

Comment:
Is this comment needed in the guide example? Will it be confusing to readers?

Reply:
Good question, it's meant for us to make it easier to replace the whole codebase at once, #18158, #25149.